File: LeetCode/C#/121. Best Time to Buy and Sell Stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        var purchasePrice = prices[0];
        var profit = 0;
        
        for(var i = 1; i < prices.Length; i++)
        {
            if(prices[i] < purchasePrice)
            {
                purchasePrice = prices[i];
            }
            else if(prices[i] - purchasePrice > profit)
            {
                profit = prices[i] - purchasePrice;
            }
        }
        
        return profit;
    }
}

File: LeetCode/Top Interview 150 - Java/121. Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = prices[0];
        int maxProfit = 0;
        int currentDiff = 0;

        for(int i = 1; i < prices.length; i++) {
            if(prices[i] < minPrice) {
                minPrice = prices[i];
            }
            else if(maxProfit < (currentDiff = prices[i] - minPrice)) {
                maxProfit = currentDiff;
            }
        }

        return maxProfit;
    }
}